conda/plugins/__init__.py

```diff
@@ -1,3 +1,4 @@
 # Copyright (C) 2012 Anaconda, Inc
 # SPDX-License-Identifier: BSD-3-Clause
 from .hookspec import hookimpl  # noqa: F401
+from .types import CondaVirtualPackage, CondaSubcommand  # noqa: F401
```

Comment:
FWIW this is the thing that makes me worried, that I mentioned in our conversations: `conda.plugins.types` will eventually have non-trivial functionality and import stuff from elsewhere in conda or even 3rd party library when we add new plugin hooks. Importing the types in the `conda.plugins` may then trigger import-time side effects that are not in line with the need to have a fast-loading, dependency-free plugin system.

@dholt You have a lot of experience with Python's import system, what's your take on this?

Reply:
Checking conda's init https://github.com/conda/conda/blob/main/conda/__init__.py loaded as a side effect of importing conda dot anything...

I'm not sure how complex types will get but agree it seems like a good idea to avoid conda.models

Reply:
Thanks for the input, let's go ahead then.